File: packages/android-sdk/sdk/src/main/kotlin/com/opencom/sdk/OpencomUser.kt
package com.opencom.sdk

/**
 * User identification data for Opencom.
 *
 * @property userId Unique identifier for the user in your system
 * @property email User's email address
 * @property name User's display name
 * @property company User's company name
 * @property customAttributes Additional custom attributes for segmentation
 */
data class OpencomUser(
    val userId: String? = null,
    val email: String? = null,
    val name: String? = null,
    val company: String? = null,
    val customAttributes: Map<String, Any> = emptyMap()
) {
    init {
        require(userId != null || email != null) {
            "Either userId or email must be provided"
        }
    }
}

File: packages/ios-sdk/Sources/OpencomSDK/Core/OpencomUser.swift
import Foundation

/// User identification data for the Opencom SDK.
public struct OpencomUser {
    /// Unique identifier for the user in your system
    public let userId: String?

    /// User's email address
    public let email: String?

    /// User's display name
    public let name: String?

    /// User's company name
    public let company: String?

    /// Custom attributes for segmentation and personalization
    public let customAttributes: [String: Any]?

    public init(
        userId: String? = nil,
        email: String? = nil,
        name: String? = nil,
        company: String? = nil,
        customAttributes: [String: Any]? = nil
    ) {
        self.userId = userId
        self.email = email
        self.name = name
        self.company = company
        self.customAttributes = customAttributes
    }
}
